### ISSUE TITLE
v2.1.1 ERR | 1 replicas failing with "could not find valid split key" (auto solved)

### ISSUE THREAD

**Issue by dendon** [cockroachdb/cockroach#32761]
I tried to add a second node to move everything to the new node and then drain the old one.

The new node shows:
```
W181201 09:17:16.565087 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:18:16.556130 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:19:16.547970 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:20:16.552979 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:21:16.600095 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:22:16.549566 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:23:16.556045 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
E181201 09:23:39.630820 158431527 storage/queue.go:893  [n2,split] 1 replicas failing with "could not find valid split key"
W181201 09:24:16.574644 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:25:16.614973 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:26:16.568024 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:27:16.549550 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:28:16.557335 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
W181201 09:29:16.550365 257 server/node.go:886  [n2,summaries] health alerts detected: {Alerts:[{StoreID:2 Category:METRICS Description:queue.split.process.failure Value:1}]}
```

The old node shows:
```
W181201 09:19:36.778645 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:20:35.574167 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:21:24.133601 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:22:12.076182 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:23:15.983628 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:24:29.917579 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:25:29.410834 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:26:44.254605 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:27:42.091767 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:28:47.820922 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:29:34.554373 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:30:37.232823 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
W181201 09:31:41.206191 377 storage/store_rebalancer.go:227  [n1,s1,store-rebalancer] StorePool missing descriptor for local store
```

The UI shows the new node has all ranges but one.

This is again happening on a production system and since I am not sure what information is needed, please let me know, but I might need to sanitize certain things.

**Comment by dendon** [comment]
Hey,
I am sorry, it seems by now to have solved on it's own.

**Comment by tbg** [comment]
Hi @dendon, glad this issue solved itself, but let's see if we can figure out what that was.

The reason this message typically comes up is because a single range is experiencing lots of writes to a single key. For example, if you updated the same row over and over (and this row were kind of large), this problem could occur.

Please let us know if that happens in your workload. If it does, and you provide some detail, we may be able to suggest a mitigation, such as introducing column families that separate the frequently updated part of a row from the rest.

**Comment by dendon** [comment]
Hello @tbg and thanks for the response.

To give a little background: I am developing and running my own CMS and am not aware of any hot/tight loop which would write a single row all the time, the shortest time some rows get updated is roughly every 30 minutes. But it is pretty likely that some ranges are updated a few times per minute, if not per second considering there are a about a hundred inserts/updates all the time to some tables.

I am also using Cockroach now for 1 1/2 years now, so I have experienced quite a bit of what it does under the hood by now.
Actually, I placed every column in its own family (since I noticed they are immutable, which is also a different feature request ticket I created some time ago).
Also, I would guess that my usage sometimes gets Cockroach to the limits of what it was made for, not performance, but feature wise, as the current problem raised when moving the whole database from one server to another which is not something you guys list as a feature ;)

My only suggestion would be to change the message to something that looks less final, like appending: trying again later . . . (as of now, the message re-occurs, yet it is not possible to distinguish if it is just hitting it again or failed while trying again)
or something in that manner, so you know something failed, but maybe not forever, because the most errors I saw by now in the production system always meant I found a bug and or an outage sooner or later or having the re-import the whole db.


Sorry if I got a bit off-topic.
